Add contributor ids field into collection summary model, and update it appropriately.

rainehoover opened the pull request:
Questions for reviewers:
- Not sure if more tests are needed? The collection summary functions do not have many tests currently (under the TODOs in that file).
- We only want contributor ids to have people who have made a "positive contribution to a collection's content." I believe the only place where a collection is edited but its content is not is in rights manager. I don't think that currently reverts are supported for collections. Let me know if I have missed this!

seanlip commented:
Looks good, except that we should replicate the changes in [Kerry's commit](https://github.com/oppia/oppia/commit/46869d0) (as discussed in email). Thanks @rainehoover!

With regards to your questions:
  - Good question. Would you mind adding a test for the case where user A creates, user B edits, then user A adds user C as an owner and D as an editor (and check that only A, B are counted in the contributor list)? That should cover the major cases.
  - You're right that reverts aren't currently supported for collections (/cc @BenHenning). I think what you've done there is fine -- when the time comes to add this, I imagine the developer will refer to the method in exp_services and follow that.

rainehoover commented:
@seanlip, added the test requested and updated the commit specified, let me know how it looks! ran the individual tests and they passed, currently running whole testing suite but wanted to save time. will chat you and update here when it finishes!

seanlip commented:
Looks good, thanks! Just small comments. Don't forget to update the global test count as well.

rainehoover commented:
Backend tests passed, all review comments addressed. Thanks @seanlip!
